pil-analyzer/tests/condenser.rs

```diff
@@ -243,3 +243,124 @@ fn new_fixed_column_as_closure() {
     "#;
     analyze_string::<GoldilocksField>(input);
 }
+
+#[test]
+fn set_hint() {
+    let input = r#"
+    namespace std::prover;
+        let set_hint = 8;
+        let eval = 8;
+        enum Query { Hint(fe), None, }
+    namespace N(16);
+        let x;
+        let y;
+        std::prover::set_hint(y, |i| std::prover::Query::Hint(std::prover::eval(x)));
+        {
+            let z;
+            std::prover::set_hint(z, query |_| std::prover::Query::Hint(1));
+        };
+    "#;
+    let expected = r#"namespace std::prover;
+    let set_hint = 8;
+    let eval = 8;
+    enum Query {
+        Hint(fe),
+        None,
+    }
+namespace N(16);
+    col witness x;
+    col witness y(i) query std::prover::Query::Hint(std::prover::eval(N.x));
+    col witness z(_) query std::prover::Query::Hint(1);
+"#;
+    let formatted = analyze_string::<GoldilocksField>(input).to_string();
+    assert_eq!(formatted, expected);
+}
+
+#[test]
+#[should_panic = "Expected type: int -> std::prover::Query"]
+fn set_hint_invalid_function() {
+    let input = r#"
+    namespace std::prover;
+        let set_hint = 8;
+        let eval = 8;
+        enum Query { Hint(fe), None, }
+    namespace N(16);
+        let x;
+        std::prover::set_hint(x, query |_, _| std::prover::Query::Hint(1));
+    "#;
+    analyze_string::<GoldilocksField>(input);
+}
+
+#[test]
+#[should_panic = "Array elements are not supported for std::prover::set_hint (called on N.x[0])."]
```

Comment:
will it be supported eventually?

Reply:
Only once we remove the old syntax, because you would have to set a hint for the whole array and we don't have syntax for that now.